src/Core/Models/Commands/CommandResult.cs

```diff
@@ -10,3 +12,51 @@ public CommandResult(string error) : this([error]) { }
 
     public CommandResult() : this(Array.Empty<string>()) { }
 }
+
+public class FailureCommandResult : CommandResult
+{
+    public FailureCommandResult(IEnumerable<string> errorMessages) : base(errorMessages)
+    {
+
+    }
+    public FailureCommandResult(string errorMessage) : base(errorMessage)
+    {
+
+    }
+}
+
+public class SuccessCommandResult : CommandResult
+{
+}
+
+public abstract class CommandResult<T>
+{
+
+}
+
+public class SuccessCommandResult<T> : CommandResult<T>
```

Comment:
I would saw we could shorten this to just Success and Failure.  Adding command result to the end is a bit of unneeded visual nois

Reply:
Should we still keep `NoRecordFoundFailure` and similar classes, though?

Reply:
Yeah. Those are probably okay because we wanna differentiate between it and the common HttpResult types.